randovania/game_description/requirements/resource_requirement.py

```diff
@@ -20,15 +19,27 @@
     from randovania.game_description.resources.simple_resource_info import SimpleResourceInfo
 
 
-@dataclasses.dataclass(frozen=True, slots=True)
 class ResourceRequirement(Requirement):
+    __slots__ = ("resource", "amount", "negate")
     resource: ResourceInfo
     amount: int
     negate: bool
 
     def __post_init__(self) -> None:
         assert TypeError("No ResourceRequirement should be directly created")
 
+    def __copy__(self) -> ResourceRequirement:
+        return type(self)(self.resource, self.amount, self.negate)
+
+    def __reduce__(self) -> tuple[type[ResourceRequirement], tuple[ResourceInfo, int, bool]]:
+        return type(self), (self.resource, self.amount, self.negate)
+
+    def __init__(self, resource: ResourceInfo, amount: int, negate: bool):
+        self.resource = resource
+        self.amount = amount
+        self.negate = negate
+        self.__post_init__()
```

Comment:
I'm definitely missing some knowledge: Is the assertion to an object not always true? If that's the case, what is the purpose of this?